File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.c
#include <stdio.h>

int minOperations(int* nums, int numsSize, int x) {
    int total = 0;
    for (int i = 0; i < numsSize; i++) {
        total += nums[i];
    }

    int target = total - x;

    // Edge cases
    if (target < 0) {
        return -1;
    }
    if (target == 0) {
        return numsSize;
    }

    int left = 0;
    int sum = 0;
    int maxLen = -1;

    for (int right = 0; right < numsSize; right++) {
        sum += nums[right];

        // Shrink window if current sum exceeds target
        while (left <= right && sum > target) {
            sum -= nums[left];
            left++;
        }

        if (sum == target) {
            int currentLen = right - left + 1;
            if (currentLen > maxLen) {
                maxLen = currentLen;
            }
        }
    }

    return (maxLen == -1) ? -1 : (numsSize - maxLen);
}

File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.py
class Solution:
    def minOperations(self, nums: list[int], x: int) -> int:
        n = len(nums)
        total = sum(nums)
        target = total - x

        # Edge cases
        if target < 0:
            return -1
        if target == 0:
            return n

        left = 0
        current_sum = 0
        max_len = -1

        for right in range(n):
            current_sum += nums[right]

            # Shrink window if current sum exceeds target
            while left <= right and current_sum > target:
                current_sum -= nums[left]
                left += 1

            if current_sum == target:
                max_len = max(max_len, right - left + 1)

        return -1 if max_len == -1 else n - max_len